src/main/java/taskat4/Main.java

```diff
@@ -0,0 +1,51 @@
+package taskat4;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Main {
```

Comment:
Demo is quite representative, but you need to correct inner classes that are used in demo in order to have correct results here